fields/types/number/NumberFilter.js

```diff
@@ -10,92 +10,108 @@ const MODE_OPTIONS = [
 	{ label: 'Between', value: 'between' },
 ];
 
-var NumberFilter = React.createClass({
+function getDefaultValue () {
+	return {
+		mode: MODE_OPTIONS[0].value,
+		value: '',
+	};
+}
 
-	getInitialState () {
+var NumberFilter = React.createClass({
+	statics: {
+		getDefaultValue: getDefaultValue,
+	},
+	getDefaultProps () {
 		return {
-			modeValue: MODE_OPTIONS[0].value, // 'matches'
-			modeLabel: MODE_OPTIONS[0].label, // 'Matches'
-			value: '',
-			minValue: '',
-			maxValue: '',
+			filter: getDefaultValue(),
 		};
 	},
 
 	componentDidMount () {
 		// focus the text input
-		ReactDOM.findDOMNode(this.refs.input).focus();
+		findDOMNode(this.refs.focusTarget).focus();
```

Comment:
* Why do we use string refs in new code?
* With React 15, findDOMNode is not needed on DOM refs, they already are the DOM node
* Would be nicer to put this in a `focusSelf()` method since it's used twice